modules/doCheckIn.js

```diff
@@ -0,0 +1,47 @@
+const fetch = require('isomorphic-fetch');
+const endpoints = require('../helpers/endpoints.js'); //For Creating shorter URL's in this Module
+const sorter = require('../helpers/sorter.js'); //For easier Sorting of various Responses.
+const objs = require('../helpers/objects.js'); //For Storing the Objects that the Framework returns. 
+const { getConfig } = require('../index');
+
+/**
+ * Loads Favorite Members
+ * @param {SecurityString} sid For authenticating with the Narvii-API.
+ * @param {String} coms A list of communities to get the check-in status, separated by "&"
+ * @param {String} timezone The desired timezone to check(It will affect the user Check-In Functionality)
+ * @returns {Object} Object with all the Check In info of the communities specified from the Logged-in User has in an Array.
+ */
+
+module.exports = async function doCheckIn(com, timezone) {
+    let checkInData = {
+        "data": {},
+        "status": "not ok",
+        "error": "nothing happening"
+    };
+    const sid = getConfig('sid');
+    if (typeof sid != 'string' || typeof com !== 'string' || typeof timezone !== 'string') {
+        throw new Error('All Arguments are not satisfied.');
+    }
+    try{
+        const response = await fetch(endpoints.doCheckIn(com), {
+            headers: {
+                'NDCAUTH': `sid=${sid}`,
+                'NDC-MSG-SIG': 'AcYt8HVzM3r6BgVIRJC+Cb4bm35F'
+            },
+            body: JSON.stringify({
+                'timezone': timezone,
+                'timestamp': new Date().getUTCMilliseconds(),
```

Comment:
Due to the limitation and bugs of the native JavaScript date time API, I highly suggest that we use Moment.js

Reply:
But i dont want AminoJS to be flooded with depenencies. I would like it to be as minimal as possible. As long as there a no errors with the native API it could be used.

Reply:
We can keep the native for now. It is working pretty well for now.
If it starts causing bugs on AminoJS, we migrate to Moment.JS

Reply:
@moelrobii Yeah u right u right, but from my past experiences that sometimes the date time API can actual costs some unexpected bugs because it will return the wrong output.

And @akatsukilevi now just thinking of MomentJS might be a bit overkill for AminoJS since we are just going to it for getting the "correct" timestamp, we can just find another alternative beside MomentJS

Reply:
As long there are no bugs in the actual system we will be using JS native.